### PR TITLE
Documentation changes for new tools modifications for ARMC5/ARMC6 usage

### PR DESCRIPTION
**Pull request by SenRamakri**
* DO NOT MERGE *
This is documentation to go with new tools changes being implemented for supporting
ARMC5 with mbed-os tools. 

This change depends on tools changes being implemented by @bridadan which are yet to be 
landed in PR form.  But please start the review process for this documentation change so that we are ready to go when tools changes are merged.

Tagging: @bridadan @bulislaw @AnotherButler @iriark01 @screamerbg @ChiefBureaucraticOfficer @kegilbert

### CODE REVIEW

File: docs/reference/configuration/mbed_targets.md
Comment (bridadan) [inline docs/reference/configuration/mbed_targets.md]: Missing a space here:
```suggestion
be the default ARM Compiler for Mbed OS. So, for a target which specifies `ARM` in `supported_toolchains`, the system will default to using `Arm Compiler 6` when you specify `-t ARM` with `mbed compile` command. If the build system
```

File: docs/tools/tools_intro.md
Comment (bridadan) [inline docs/tools/tools_intro.md]: Are sure about that date? I was told to not put any dates in the warning that the tools will be printing.

File: docs/reference/configuration/mbed_targets.md
Comment (bridadan) [inline docs/reference/configuration/mbed_targets.md]: Good to know, I'm not currently erroring on this 😄. I can make this change.

File: docs/reference/configuration/mbed_targets.md
Comment (bridadan) [inline docs/reference/configuration/mbed_targets.md]: I suggest being more definite when documenting the behavior of the tools:
```suggestion
fails to detect a valid configuration for `Arm Compiler 6`, it will automatically(without any manual intervention) try to detect if a valid configuration for `ARM Compiler 5` exists, and will continue using `ARM Compiler 5` if it successfully detects `ARM Compiler 5`. You will see a warning message in your standard output from the build system indicating this behavior. 
```

**Review by bridadan — CHANGES_REQUESTED** [review]
Already looking quite good, a few suggestions:

File: docs/tools/tools_intro.md
Comment (SenRamakri) [reply docs/tools/tools_intro.md]: @AnotherButler - What do you think?

Comment (AnotherButler) [reply docs/tools/tools_intro.md]: That was requested specifically by Product this afternoon.

File: docs/reference/configuration/mbed_targets.md
Comment (AnotherButler) [inline docs/reference/configuration/mbed_targets.md]: If the target supports Arm Compiler 5, will `supported_toolchains` show `ARMC5`, or do you have to manually add that?

File: docs/tools/tools_intro.md
Comment (SenRamakri) [reply docs/tools/tools_intro.md]: @AnotherButler @bridadan - I think this needs to be clarified between Product and TAM as I hear different things from them. If we fail to reach an agreement we should go back to the original message I added.

File: docs/reference/configuration/mbed_targets.md
Comment (SenRamakri) [reply docs/reference/configuration/mbed_targets.md]: @AnotherButler - Good question, the changes which Brian is working on will add everything as required and according to these rules. So no manual addition required from user or the reader of this documentation. But I said "If you specify" because this document is talking about "Adding and configuring targets".

Comment (AnotherButler) [reply docs/reference/configuration/mbed_targets.md]: So if I changed it to "If the target supports Arm Compiler 5 (version 5.06 update 6) only, `supported_toolchains` will specify `ARMC5`", would that be correct?

Comment (SenRamakri) [reply docs/reference/configuration/mbed_targets.md]: @AnotherButler - It is technically correct. But I think the intended audience for this document is someone who is adding or configuring a target, as in the document title. That's why its written as "if you specify.." meaning whoever is adding or configuring a target. But I also want that information to be available for anyone, so I created a link from cli-compile.md to here. So, Ill leave that up to you and if the new format is cleaner/compact I would keep that.

**Review by kegilbert — APPROVED** [review]

**Review by bridadan — APPROVED** [review]
Edits looking real nice!

**Comment by bridadan** [comment]
What would the URL for this doc be once it goes live? I will need to add this link to my tools PR. Thanks!

**Comment by bridadan** [comment]
Also, do we have docs telling the user how to upgrade to ARMC6? Should that be part of this doc or in a separate place?

**Comment by iriark01** [comment]
> What would the URL for this doc be once it goes live? I will need to add this link to my tools PR. Thanks!

https://os.mbed.com/docs/mbed-os/v5.12/reference/configuration.html
https://os.mbed.com/docs/mbed-os/v5.12/tools/compile.html
https://os.mbed.com/docs/mbed-os/v5.12/tools/index.html

File: docs/tools/tools_intro.md
Comment (iriark01) [reply docs/tools/tools_intro.md]: I'll try to get it sorted with product.

**Review by 0xc0170 — APPROVED** [review]

File: docs/reference/configuration/mbed_targets.md
Comment (bulislaw) [inline docs/reference/configuration/mbed_targets.md]: Can we add a note here saying that it's free for Mbed usecases and link to instructions how the get it?

File: docs/tools/tools_intro.md
Comment (bulislaw) [inline docs/tools/tools_intro.md]: Can we add a note saying it's free for Mbed users and link to the instructions?

**Review by bulislaw — CHANGES_REQUESTED** [review]

File: docs/reference/configuration/mbed_targets.md
Comment (iriark01) [reply docs/reference/configuration/mbed_targets.md]: I can do that - do you happen to know where those instructions are? I'm not familiar with this bit.

Comment (iriark01) [reply docs/reference/configuration/mbed_targets.md]: Ah, found something: https://developer.arm.com/products/software-development-tools/compilers/arm-compiler/downloads/version-6

Comment (iriark01) [inline docs/reference/configuration/mbed_targets.md]:
```suggestion
<span class="notes">**Note**: Arm Compiler 6 is the default Arm Compiler version for Mbed OS development, and is [free for use with Mbed OS]((https://os.mbed.com/docs/mbed-os/development/tools/index.html#compiler-versions). Most platforms are already compatible with it; platforms still supporting Arm Compiler 5 will be migrated to Arm Compiler 6. Please do not use Arm Compiler 5 in any new development, as its support will be deprecated in September 2019.
.</span>
```

File: docs/tools/tools_intro.md
Comment (iriark01) [inline docs/tools/tools_intro.md]:
```suggestion
<span class="notes">**Note**: Arm Compiler 6 is the default Arm Compiler version for Mbed OS development, and is [free for use with Mbed OS]((https://os.mbed.com/docs/mbed-os/development/tools/index.html#compiler-versions). Most platforms are already compatible with it; platforms still supporting Arm Compiler 5 will be migrated to Arm Compiler 6. Please do not use Arm Compiler 5 in any new development, as its support will be deprecated in September 2019.
.</span>
```

Comment (iriark01) [reply docs/tools/tools_intro.md]: Mohit says September is necessary to reassure users we aren't violating our "six months warning" deprecation policy.

Comment (iriark01) [reply docs/tools/tools_intro.md]: @SenRamakri this is a fork, not a branch. I therefore don't have permission to push these suggestions. Please push if you approve them.

File: docs/reference/configuration/mbed_targets.md
Comment (iriark01) [reply docs/reference/configuration/mbed_targets.md]: @SenRamakri this is a fork, not a branch. I therefore don't have permission to push these suggestions. Please push if you approve them.

**Comment by SenRamakri** [comment]
@iriark01 - I have pushed your suggestions, please take a look.

File: docs/tools/tools_intro.md
Comment (iriark01) [inline docs/tools/tools_intro.md:46]: My mistake, in both notes, there are two `((` in the link; there should be only one

File: docs/reference/configuration/mbed_targets.md
Comment (iriark01) [inline docs/reference/configuration/mbed_targets.md:209]: My mistake, in both notes, there are two `((` in the link; there should be only one

**Comment by iriark01** [comment]
Tried to resolve the conflict but I don't know which version is actually correct. @SenRamakri please can you resolve those? And @bulislaw can you recheck and approve?

**Comment by SenRamakri** [comment]
@iremic - I have resolved the merge conflicts as well. Thanks much for making those changes.

**Comment by iriark01** [comment]
Okay; I'll give Bartek time to dismiss his review or ask for more changes. Other than that we're good to go (I can fix the funky links when it's merged)

**Review by bulislaw — APPROVED** [review]
Looks good. One thing about page we are linking to tell users about free armc6 and how to get/confiugre it https://os.mbed.com/docs/mbed-os/development/tools/index.html#compiler-versions there's nothing there about free armc6 for mbed studio and how to use it with mbed cli.

**Comment by iriark01** [comment]
Do we have that content anywhere? Do we need to write it? 
(Full Mbed Studio docs here: https://os.mbed.com/docs/mbed-studio/)

**Comment by iriark01** [comment]
Amanda said we have content in the release note; I'll see if I can get anything sensible into the docs from that

**Comment by iriark01** [comment]
I'm merging this so we can see everything in one place (this is a fork, not a branch, so we don't get a preview on the docs site)

**Pull request merged**
